app/Connect/Commands/AuthenticatedApiHandlerBase.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Connect\Commands;
+
+use App\Models\User;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Http\Request;
+
+abstract class AuthenticatedApiHandlerBase extends ApiHandlerBase
+{
+    public User $user;
+
+    public function execute(Request $request): JsonResponse
+    {
+        if (!$this->authenticate($request)) {
+            return $this->buildResponse($this->accessDenied());
+        }
+
+        return parent::execute($request);
+    }
```

Comment:
An `AuthenticatedApiHandlerBase` extends the `ApiHandlerBase` to validate the user's credentials before doing anything else.